pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -350,7 +351,8 @@ private StateStoreProvider getStateStoreProvider() throws Exception {
         }
     }
 
-    private void handleResult(Record srcRecord, JavaExecutionResult result) throws Exception {
+    @VisibleForTesting
+    protected void handleResult(Record srcRecord, JavaExecutionResult result) throws Exception {
```

Comment:
would package private be more suitable?

Reply:
I think it's fine to do that, but package private is more restrictive than protected but still gives the access you need for testing.